Sources/JavaKit/generated/JavaOptionalLong.swift

```diff
@@ -0,0 +1,44 @@
+// Auto-generated by Java-to-Swift wrapper generator.
+import JavaRuntime
+
+@JavaClass("java.util.OptionalLong")
+open class JavaOptionalLong: JavaObject {
+  @JavaMethod
+  open override func equals(_ arg0: JavaObject?) -> Bool
+
+  @JavaMethod
+  open override func toString() -> String
+
+  @JavaMethod
+  open override func hashCode() -> Int32
+
+  @JavaMethod
+  open func isEmpty() -> Bool
+
+  @JavaMethod
+  open func isPresent() -> Bool
+
+  @JavaMethod
+  open func orElse(_ arg0: Int64) -> Int64
+
+  @JavaMethod
+  open func orElseThrow() -> Int64
+
+  @JavaMethod
+  open func getAsLong() -> Int64
+}
+extension JavaClass<JavaOptionalLong> {
+  @JavaStaticMethod
+  public func of(_ arg0: Int64) -> JavaOptionalLong!
+
+  public func ofOptional(_ arg0: Int64) -> Int64? {
+    Optional(javaOptional: of(arg0))
+  }
+
+  @JavaStaticMethod
+  public func empty() -> JavaOptionalLong!
+
+  public func emptyOptional() -> Int64? {
+    Optional(javaOptional: empty())
+  }
```

Comment:
Not sure I understand where the ofOptional and emptyOptional came from? I know this is generated but couldn't quite wrap my head around what the intent of these is?

Reply:
This is a fun one:
Any method that returns a Java optional (or friends) now has a second method that returns a swift optional and does wrapping. A fun consequence of this is that all optionals have static methods to create themselves ... which then have a helper generated to make a swift optional. It's a fun bit of recursion-ish stuff